ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/TableSuppressBuilderTest.java

```diff
@@ -0,0 +1,132 @@
+import org.apache.kafka.streams.processor.StateStore;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class TableSuppressBuilderTest {
+
+  @Mock
+  private KsqlQueryBuilder queryBuilder;
+  @Mock
+  private ExecutionStep<KTableHolder<Struct>> sourceStep;
+  @Mock
+  private KTable<Struct, GenericRow> sourceKTable;
+  @Mock
+  private KTable<Struct, GenericRow> preKTable;
+  @Mock
+  private KTable<Struct, GenericRow> suppressedKTable;
+  @Mock
+  private RefinementInfo refinementInfo;
+  @Mock
+  private Formats internalFormats;
+  @Mock
+  private KsqlWindowExpression windowExpression;
+  @Mock
+  private KeySerdeFactory<Struct> keySerdeFactory;
+  @Mock
+  private MaterializedFactory materializedFactory;
+  @Mock
+  private  PhysicalSchema physicalSchema;
+  @Mock
+  private  Serde<GenericRow> valueSerde;
+  @Mock
+  private  Serde<Struct> keySerde;
+  @Mock
+  private  Materialized<Object, GenericRow, StateStore> materialized;
+  @Mock
+  private KTableHolder<Struct> tableHolder;
+  @Mock
+  private KTableHolder<Struct> suppressedtable;
+
+  private final QueryContext queryContext = new QueryContext.Stacker()
+      .push("bar")
+      .getQueryContext();
+  private TableSuppress<Struct> tableSuppress;
+  private BiFunction<LogicalSchema, Set<SerdeOption>, PhysicalSchema> physicalSchemaFactory;
+  private TableSuppressBuilder builder;
+
+  @Rule
+  public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Before
+  @SuppressWarnings("unchecked")
+  public void init() {
+    final ExecutionStepPropertiesV1 properties = new ExecutionStepPropertiesV1(queryContext);
+
+    physicalSchemaFactory = (a,b) -> physicalSchema;
+    when(queryBuilder.buildValueSerde(any(), any(), any())).thenReturn(valueSerde);
+    when(keySerdeFactory.buildKeySerde(any(), any(), any())).thenReturn(keySerde);
+    when(materializedFactory.create(any(), any(), any())).thenReturn(materialized);
+
+    when(tableHolder.getTable()).thenReturn(sourceKTable);
+    when(sourceKTable.transformValues(any(), any(Materialized.class))).thenReturn(preKTable);
+    when(preKTable.suppress(any())).thenReturn(suppressedKTable);
+    when(tableHolder.withTable(any(),any())).thenReturn(suppressedtable);
+
+    tableSuppress = new TableSuppress<>(properties, sourceStep, refinementInfo, internalFormats, windowExpression);
+    builder = new TableSuppressBuilder();
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void shouldSuppressSourceTable() {
+    // When:
+    final KTableHolder<Struct> result = builder.build(tableHolder, tableSuppress, queryBuilder, keySerdeFactory, materializedFactory, physicalSchemaFactory);
+
+    // Then:
+    assertThat(result, is(suppressedtable));
+    verify(sourceKTable).transformValues(any(),any(Materialized.class));
+    verify(preKTable).suppress(any());
```

Comment:
it might be valuable to ensure that we're calling `suppress` with a `FinalResultsSuppressionBuilder` passed in. You can see usages of `@Captor` to see examples of how to use mockito to intercept the arguments being passed in and very that they're of the type you expect.

This is in contrast to other supressed instances we might be passing in in the future if we support things other than window close.